cpp/include/Ice/ClientAuthenticationOptions.h

```diff
@@ -0,0 +1,414 @@
+     */
+    struct ClientAuthenticationOptions
+    {
+#if defined(_WIN32)
+        /**
+         * A callback that allows selecting the client's SSL certificate based on the target server host name.
+         *
+         * @remarks This callback is invoked by the SSL transport for each new outgoing connection before starting the
+         * SSL handshake to determine the appropriate client certificate. The callback should return a PCCERT_CONTEXT
+         * that represents the client's certificate. The SSL transport takes ownership of the returned certificate and
+         * releases it when the connection is closed.
+         *
+         * @param host The target server host name.
+         * @return The client's certificate, or nullptr to indicate that no certificate is used by the connection.
+         *
+         * Example of setting clientCertificateSelectionCallback:
+         * ```cpp
+         * PCCERT_CONTEXT _clientCertificate = ...; // Load the client certificate using WinCrypt API
+         *
+         * auto initData = Ice::InitializationData {
+         *   ...
+         *   .clientAuthenticationOptions = ClientAuthenticationOptions {
+         *      .clientCertificateSelectionCallback = [this](const std::string&)
+         *      {
+         *        // Increment the certificate context reference count to ensure it remains
+         *        // valid for the duration of the connection. The SSL transport will release
+         *        // it after closing the connection.
+         *        CertDuplicateCertificateContext(_clientCertificate);
+         *        return _clientCertificate;
+         *      }
+         * };
+         *
+         * auto communicator = Ice::initialize(initData);
+         * ...
+         * CertFreeCertificateContext(_clientCertificate); // Release the certificate when no longer needed
+         * ```
+         *
+         * See Detailed Wincrypt documentation for [CERT_CONTEXT](
+         * https://learn.microsoft.com/en-us/windows/win32/api/wincrypt/ns-wincrypt-cert_context) and
+         * [CertCreateCertificateContext](https://learn.microsoft.com/en-us/windows/win32/api/wincrypt/nf-wincrypt-certcreatecertificatecontext)
+         */
+        std::function<PCCERT_CONTEXT(const std::string& host)> clientCertificateSelectionCallback;
+
+        /**
+         * The trusted root certificates used for validating the server's certificate chain. If this field is set, the
+         * server's certificate chain is validated against these certificates; otherwise, the system's default root
+         * certificates are used.
+         *
+         * @remarks The trusted root certificates are used by both the default validation callback, and by custom
+         * validation callback set in clientCertificateValidationCallback.
+         *
+         * The application must ensure that the certificate store remains valid during the setup of the Communicator. It
```

Comment:
Should setup be initialization? 
```suggestion
         * The application must ensure that the certificate store remains valid during the setup of the Communicator. It
```

Reply:
I think that's actually obvious and not necessary to document. initialize() is a synchronous function. While it runs, you should not free the arguments you passed via initData.

Reply:
What about the second part about releasing the certificate store?